src/alert.js

```diff
@@ -0,0 +1,64 @@
+// A Growl-style alert that slides into view at the top of the screen when rendered,
+// and slides back out of view when the "X" button is clicked/pressed.
+// Visit The Stimulus Handbook for more details
+// https://stimulusjs.org/handbook/introduction
+//
+// This example controller works with specially annotated HTML like:
+//
+//  <div class="fixed inset-x-0 top-0 flex items-end justify-right px-4 py-6 sm:p-6 justify-end z-50 ">
+//     <div data-controller="alert" class="max-w-sm w-full shadow-lg rounded px-4 py-3 rounded relative bg-green-400 border-l-4 border-green-700 text-white">
+//       <div class="p-2">
+//         <div class="flex items-start">
+//           <div class="ml-3 w-0 flex-1 pt-0.5">
+//             <p class="text-sm leading-5 font-medium">
+//               Stimulus is the JS of the future!
+//             </p>
+//           </div>
+//           <div class="ml-4 flex-shrink-0 flex">
+//             <button data-action="alert#close" class="inline-flex text-white focus:outline-none focus:text-gray-300 transition ease-in-out duration-150">
+//               <svg class="h-5 w-5" viewBox="0 0 20 20" fill="currentColor">
+//                 <path fill-rule="evenodd" d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z" clip-rule="evenodd"/>
+//               </svg>
+//             </button>
+//           </div>
+//         </div>
+//       </div>
+//     </div>
+//   </div>
+
+import { Controller } from 'stimulus'
+
+export default class extends Controller {
+
+  initialize() {
+    this.hide()
+  }
+
+  connect() {
+    setTimeout(() => {
+      this.show()
+    }, 200)
+  }
```

Comment:
Does it make sense to also have a `disconnect()` handler? That would get triggered as the element gets removed from the DOM which could be either the alert closing itself or navigating away.

I'm wondering if we'd want to `clearTimeout` in there to clear the timeouts?

Just thinking out loud. I'm fairly certain it's not necessary, but wanted to mention it in case anyone had some thoughts.

Reply:
I don't think it's necessary, since one of the timeouts governs the initial appearance of the alert, and the other governs hiding and then removing the alert.  That said, I could probably be convinced otherwise if someone feels strongly about it.

Reply:
I was thinking of the situation with Turbolinks where you navigate to another page before the alert closes. 

The timeout would still fire to remove the element, which no longer exists and I think that would cause an error if you didn't cancel the timeout, right?